src/java/org/apache/cassandra/db/compaction/unified/CostsCalculator.java

```diff
@@ -0,0 +1,279 @@
+     * <p/>
+     * The number of sstables is calculated as Math.min(1 + env.bloomFilterFpRatio() * RA / survivalFactor, RA). Here we
+     * assume there is going to be at least one sstable accessed, possibly more in case of :
+     *
+     * - bloom filter's false positives;
+     * - partitions not surviving a compaction (1/survivalFactor is the limit of the sum of (1-survivalFactor)^n), that
+     *   is partitions that would not exist if compaction was done; Note that the survival factor is currently fixed to 1.
+     *
+     * The RA is then a cap since we cannot read more than RA sstables, which are the sstables that exist because
+     * compactions allows them to exist.
+     * </p>
+     * The read multiplier is a factor that operators can use to tweak the algorithm.
+     * </p>
+     * @param RA the expected read amplification due to the current choice of compaction strategy
+     *
+     * @return the projected read cost for user queries
+     */
+    public double getReadCostForQueries(int RA)
+    {
+        readLock.lock();
+
+        try
+        {
+            return getReadCost(partitionsRead.avg.get()) * Math.min(1 + env.bloomFilterFpRatio() * RA / survivalFactor, RA) * readMultiplier;
+        }
+        finally
+        {
+            readLock.unlock();
+        }
+    }
+
+    private double getFlushCost(double bytesWritten)
+    {
+        return ((bytesWritten / (1 << 10)) * env.flushLatencyPerKbInNanos()) / (double) TimeUnit.MILLISECONDS.toNanos(1);
+    }
+
+    private double getCompactionCost(double bytesWritten)
+    {
+        // So, the compaction latency will depend on the size of the sstables, so in the correct solution each level
+        // should pass its output size and we should measure latency in MB or something like that
+        return ((bytesWritten / (1 << 10)) * env.compactionLatencyPerKbInNanos()) / (double)  TimeUnit.MILLISECONDS.toNanos(1);
+    }
+
+    /**
+     * Calculate the projected write cost for user insertions.
+     *
+     * The projected write cost is given by the number of bytes that were inserted times the flush cost
+     * plus the same number of bytes times the compaction cost and the compaction WA. We also multiply by
+     * a write multiplier to let users change the weights if needed.
+     *
+     * @param WA the expected write amplification due to compaction
+     *
+     * @return the projected flush and write cost.
+     */
+    public double getWriteCostForQueries(int WA)
+    {
+        readLock.lock();
+
+        try
+        {
+            double bytesInserted = this.bytesInserted.avg.get();
+            // using bytesInserted for the compaction cost doesn't take into account overwrites but for now it's good enough
+            return (getFlushCost(bytesInserted) + getCompactionCost(bytesInserted) * WA) * writeMultiplier;
+        }
+        finally
+        {
+            readLock.unlock();
+        }
+    }
+
+    public MovingAverage partitionsRead()
```

Comment:
Could we just return the `double` value here and below?

Reply:
Fixed in eb8736fdc05ff16cc37af790d14a8ffc61d977e1.